build-scripts/JenkinsFile

```diff
@@ -0,0 +1,676 @@
+
+            [$class: TEXT_PARAM_DEF_CLASS, description: 'Please provide Release Version to clean', name: 'ballerina-release-version']
+    ]);
+
+    nodeLabel = cleanReleaseParamInput['node-label'];
+
+    releaseVersionToClean = cleanReleaseParamInput['ballerina-release-version'];
+    releaseTagToClean = "v" + releaseVersionToClean;
+
+}
+
+failFast = YES.equals(enableFailFast);
+
+node ("${nodeLabel}") {
+    def tagMap = [];
+    String user = env.GIT_USER;
+    String gitCredentialsID = env.GIT_CREDENTIALS_ID;
+    String gitEmail = env.GIT_EMAIL;
+    String settingsXMLID = env.SETTINGS_XML_ID;
+    String ballerinaRepository = "ballerina";
+    String composerRepository = "composer";
+    String docerinaRepository = "docerina";
+    String testerinaRepository = "testerina";
+    String connectorsRepository = "connectors";
+    String containerSupportRepository = "container-support";
+    String toolsDistributionRepository = "tools-distribution";
+    String toolSwaggerBallerinaRepository = "tool-swagger-ballerina";
+    String pluginMavenRepository = "plugin-maven";
+    String ballerinaParentRepository = "ballerina-parent";
+    String ballerinaGroudId = "org.ballerinalang";
+    String ballerinaConnectorsGroupId = "org.wso2.ballerina.connectors"
+    String propertyUpdateCommitMessage = "Update versions of org.ballerinalang.* dependencies";
+    String parentUpdateCommitMessage = "Update ballerina-parent version";
+
+    try {
+        stage('Building ballerina-parent') {
+            cloneRepository(ballerinaParentRepository, user, gitCredentialsID, gitEmail);
+            if (isRelease) {
+                tagMap.add([ballerinaParentRepository, ballerinaParentReleaseTag, ballerinaParentVersion]);
+                release(settingsXMLID, ballerinaParentRepository, ballerinaParentVersion, ballerinaParentReleaseTag, ballerinaParentDevVersion, YES);
+            }
+
+            if (isBuild) {
+                build(settingsXMLID, ballerinaParentRepository, YES);
+            }
+
+            if (isCleanRelease) {
+                cleanRelease(ballerinaParentRepository, user, releaseTagToClean, releaseVersionToClean, gitCredentialsID)
+            }
+        }
+
+        stage('Building ballerina') {
+            cloneRepository(ballerinaRepository, user, gitCredentialsID, gitEmail);
+            if (isRelease) {
+                updateParent(ballerinaRepository, settingsXMLID, parentUpdateCommitMessage, !isRelease);
+                tagMap.add([ballerinaRepository, ballerinaReleaseTag, ballerinaVersion]);
+                release(settingsXMLID, ballerinaRepository, ballerinaVersion, ballerinaReleaseTag, ballerinaDevVersion, ballerinaSkipTests);
+                //Following two steps are for updating ballerina-parent to latest snapshot
+                updateParent(ballerinaRepository, settingsXMLID, parentUpdateCommitMessage, true);
+                //build(settingsXMLID, ballerinaRepository, YES);
```

Comment:
Will remove these as I've added building the next snapshot version into the 'build' method.